src/httpserver.cpp

```diff
@@ -480,6 +475,14 @@ void StopHTTPServer()
         workQueue->WaitExit();
         delete workQueue;
     }
+    if (eventBase) {
+        LogPrint("http", "Waiting for HTTP event thread to exit\n");
+        // Give event loop a few seconds to exit (to send back last RPC responses), then break it
+        if (!threadHTTP.try_join_for(boost::chrono::milliseconds(2000))) {
```

Comment:
New connections are stopped just above, see where the accept sockets are torn down.

Though it does look like we should close connected sockets in http_reject_request_cb as well for good measure. @laanwj Is there a reason to keep them open once they've had a req rejected?